src/rgw/rgw_sal.h

```diff
@@ -271,6 +274,7 @@ class Store {
 				  const rgw_placement_rule *ptail_placement_rule,
 				  uint64_t olh_epoch,
 				  const std::string& unique_tag) = 0;
+    virtual RGWHandler_REST* get_rest_s3_handler(const rgw::auth::StrategyRegistry& auth_registry) = 0;
```

Comment:
this feels like a layering violation to me - the rest handlers and ops are supposed to be above/outside of zipper

i see how it's necessary for a rados-specific `GetObjLayout`, but that's more of an exception than the rule. instead of making the op itself specific to the rados store, can we make the object's layout generic? for example, by adding something like `virtual int get_obj_layout(Formatter* f)` to `sal::Object` whose output is just some json?

that would allow us to keep the existing `GetObjLayout` above zipper, and its response would differ by backend